### PR TITLE
ChargeItem supporting verifier for performer

### PR DESCRIPTION
**Pull request by Anurag-Balannagari**
Description
----
Adding terminology binding for performer

Screen Shot 2021-07-06 at 12 40 01 PM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/r4/charge_item.yaml [hunk 1]
```diff
@@ -105,3 +105,17 @@ fields:
         - display: Bill Code Schedule
           system: https://fhir.cerner.com/&lt;EHR source id&gt;/codeSet/14002
           info_link: https://fhir.cerner.com/millennium/r4/proprietary-codes-and-systems/#code-set-14002-bill-code-schedule
+
+  - name: performer
+    required: 'No'
+    type: performerFunction
+    description: Distinguishes the type of involvement of the performer in the event.
+    binding:
+      description: Codes identifying the function of the performer.
```
Comment (Coffeevee) [inline lib/resources/r4/charge_item.yaml:114]: Let's update the description wording to be `A code that identifies the function of the performer.`. This wording is a little more in-line with how other terminology bindings are described.

File: lib/resources/r4/charge_item.yaml [hunk 2]
```diff
@@ -105,3 +105,17 @@ fields:
         - display: Bill Code Schedule
           system: https://fhir.cerner.com/&lt;EHR source id&gt;/codeSet/14002
           info_link: https://fhir.cerner.com/millennium/r4/proprietary-codes-and-systems/#code-set-14002-bill-code-schedule
+
+  - name: performer
```
Comment (alexbeckner1) [inline lib/resources/r4/charge_item.yaml:109]: should we be more specific and call it performer.function? I know we have done that in other places like [Patient ](http://fhir.cerner.com/millennium/r4/base/individuals/patient/#terminology-bindings) for patient.identifier.type

Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/682/commits/be52c43e6f896a3e8f843c8d310a90bd9b638d09#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR109

File: lib/resources/r4/charge_item.yaml (hunk 1 above)
Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/682/commits/be52c43e6f896a3e8f843c8d310a90bd9b638d09#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR114

File: lib/resources/r4/charge_item.yaml (hunk 2 above)
Comment (alexbeckner1) [reply lib/resources/r4/charge_item.yaml]: thanks for making that change.

**Review by alexbeckner1 — APPROVED** [review]

**Review by Coffeevee — APPROVED** [review]

File: lib/resources/r4/charge_item.yaml [hunk 3]
```diff
@@ -105,3 +105,17 @@ fields:
         - display: Bill Code Schedule
           system: https://fhir.cerner.com/&lt;EHR source id&gt;/codeSet/14002
           info_link: https://fhir.cerner.com/millennium/r4/proprietary-codes-and-systems/#code-set-14002-bill-code-schedule
+
+  - name: performer.function
+    required: 'No'
+    type: performerFunction
```
Comment (exopsRamba) [inline lib/resources/r4/charge_item.yaml:111]: If we are specifying performer.function we should set the type to CodeableConcept.

File: lib/resources/r4/charge_item.yaml [hunk 4]
```diff
@@ -105,3 +105,17 @@ fields:
         - display: Bill Code Schedule
           system: https://fhir.cerner.com/&lt;EHR source id&gt;/codeSet/14002
           info_link: https://fhir.cerner.com/millennium/r4/proprietary-codes-and-systems/#code-set-14002-bill-code-schedule
+
+  - name: performer.function
```
Comment (exopsRamba) [inline lib/resources/r4/charge_item.yaml:109]: The structure we used for communication had the parent field and then the children with the terminology bindings. Do we want to do that structure here as well? https://github.com/cerner/fhir.cerner.com/blob/main/lib/resources/r4/related_person.yaml#L26

cc @alexbeckner1

Comment (alexbeckner1) [reply lib/resources/r4/charge_item.yaml:109]: I'm fine with following the same pattern as what was done on relatedPerson.communication.

Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: updated here https://github.com/cerner/fhir.cerner.com/pull/682/commits/050eba2bc5d5c77b30288e3d85fb076e792e451f#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR109-R128

File: lib/resources/r4/charge_item.yaml (hunk 3 above)
Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: Yes this is not applicable now after changes made for this below comment.
https://github.com/cerner/fhir.cerner.com/pull/682#discussion_r663179455

**Review by babukarthiks — APPROVED** [review]

File: lib/resources/r4/charge_item.yaml (hunk 4 above)
Comment (exopsRamba) [reply lib/resources/r4/charge_item.yaml]: Left some comments on the commit.

Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: responded here https://github.com/cerner/fhir.cerner.com/commit/050eba2bc5d5c77b30288e3d85fb076e792e451f#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR109-R128

**Review by exopsRamba — APPROVED** [review]

**Review by ahamdany — APPROVED** [review]

File: lib/resources/r4/charge_item.yaml [hunk 5]
```diff
@@ -105,3 +105,24 @@ fields:
...
+        action: terminology
+        binding:
+          description: A code that identifies the function of the performer.
+          terminology:
+            - display: Performer function codes
+              system: http://terminology.hl7.org/CodeSystem/v3-ParticipationType
+              info_link: http://hl7.org/fhir/R4/valueset-performer-function.html
+            - display: Participation codes
+              system: http://terminology.hl7.org/CodeSystem/v2-0912
+              info_link: http://terminology.hl7.org/CodeSystem/v2-0912
```
Comment (mayur-rajendran) [inline lib/resources/r4/charge_item.yaml:128]: can you change the info_links to use https instead of http?

**Review by mayur-rajendran — APPROVED** [review]

File: lib/resources/r4/charge_item.yaml (hunk 5 above)
Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: updated system and info_links to https here https://github.com/cerner/fhir.cerner.com/pull/682/commits/144f92e439aba8231f1986ae1675c80928711b36#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR124-R128

Comment (Anurag-Balannagari) [reply lib/resources/r4/charge_item.yaml]: revert system links back to http here https://github.com/cerner/fhir.cerner.com/pull/682/commits/1df8b1f2f90cb56f78df6812be0f33692283d5ae#diff-7ab995fcc2135db120dc94b545fa48898edd21d9d220eee5efae95920c369dcbR124-R127

**Pull request merged**